libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -852,9 +860,14 @@ void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, u
         if (deserialize_success == false) {
             break;
         }
-        mode_switch_response.status = AP::vehicle()->set_mode(mode_switch_request.mode, ModeReason::DDS_COMMAND);
-        mode_switch_response.curr_mode = AP::vehicle()->get_mode();
-
+        if (external_control->is_enabled()) {
+            mode_switch_response.status = AP::vehicle()->set_mode(mode_switch_request.mode, ModeReason::DDS_COMMAND);
+            mode_switch_response.curr_mode = AP::vehicle()->get_mode();
+        } else {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s Mode Switch Request rejected: External Control Disabled", msg_prefix);
+            mode_switch_response.status = false;
+            mode_switch_response.curr_mode = AP::vehicle()->get_mode();
```

Comment:
this common get_mode line can be outside the if()

Reply:
applied.